UI/script.js

```diff
@@ -0,0 +1,361 @@
+                    console.log('Something went wrong :(');
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    add(adItem){
+        if(AdList.validate(adItem)){
+            if (this._adList.filter(offer => offer['id'] === adItem['id']).length === 0){
+                this._adList.push(adItem);
+                return true;
+            }
+            else{
+                console.log('this offer can\'t added, because your collection containes offer with this id');
+                return false;
+            }
+        }
+        else{
+            console.log('offer isn\'t validated');
+            return false;
+        }
+    }
+
+    edit(id, adItem){
+        for(let parameteres in adItem){
+            if(parameteres === 'id' || parameteres === 'vendor' || parameteres === 'createdAt'){
+                console.log('you can\'t change id, vendor name, createdAt');
+                return false;
+            }
+        }
+        
+        let cloneOffer = {};
+        let bufOffer = this.get(id);
+        for(let key in bufOffer){
+            cloneOffer[key] = bufOffer[key];
+        }
+        for(let param in adItem){
+            cloneOffer[param] = adItem[param];
+        }
+        if(!AdList.validate(cloneOffer)){
+            console.log('new offer isn\'t validated');
+            return false;
+        }
+        
+        for (let key in cloneOffer){
+            bufOffer[key] = cloneOffer[key];
+        }
+        return true;
+    }
+
+    remove(id){
+        if (typeof id === 'string'){
+            let index = this._adList.findIndex(offer => offer.id === id);
+            if (index !== -1){
+                this._adList.splice(index, 1);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    addAll(adList){
+        let unvalidatedOffers = adList.filter(offer => !this.add(offer));
+        return unvalidatedOffers;
+    }
+
+    clear(){
+        this._adList.splice(0, this._adList.length);
+    }
+}
+
+class View {
+    _offers;
+    _username;
+    _vendorStatus;
+    _tegOffersList;
+    _offerTmp;
+
+    constructor(offerList, username, vendorStatus){
+        this._offers = new AdList(offerList);
+        this._username = username;
+        this._vendorStatus = vendorStatus;
+        this._tegOffersList = document.querySelector('.offers_list');
+        this._offerTmp = document.querySelector('.offer');
+        let deletedOffer = document.querySelector('.offer');
+        deletedOffer.remove();
+    }
+
+    showStatus(){
+        if(this._username === null || this._username.length === 0) {
+            document.querySelector('.Sign_in').textContent = 'Log in';
+            //document.getElementsByName('door')[0].style.visibility = 'hidden';
+            document.querySelector('.door').style.visibility = 'hidden';
+        }
+        else{
+            document.querySelector('.Sign_in').textContent = this._username;
+        }
+    }
```

Comment:
I guess, you should also handle hiding/showing add new ad button, add review, edit etc